diff --git a/usr/share/applications/gitkraken.desktop b/usr/share/applications/gitkraken.desktop
index a328b53..0664ebe 100644
--- a/usr/share/applications/gitkraken.desktop
+++ b/usr/share/applications/gitkraken.desktop
@@ -2,7 +2,7 @@
 Name=GitKraken
 Comment=Unleash your repo
 GenericName=Git Client
-Exec=/usr/share/gitkraken/gitkraken %U
+Exec=/usr/bin/gitkraken %U
 Icon=/usr/share/pixmaps/gitkraken.png
 Type=Application
 StartupNotify=true
