diff --git a/data/meson.build b/data/meson.build
index e5af936..a2f52f6 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -188,7 +188,7 @@ if build_daemon
     install_dir: join_paths(datadir, 'dbus-1', 'system-services'),
   )
 
-  if launchctl.found()
+  if launchctl.found() or elogind.found()
     con2 = configuration_data()
     con2.set('libexecdir', libexecdir)
     con2.set('dbus_socket_address', get_option('dbus_socket_address'))
diff --git a/meson.build b/meson.build
index 8bffa1e..7626fb5 100644
--- a/meson.build
+++ b/meson.build
@@ -517,6 +517,11 @@ if libsystemd.found()
   endif
 endif
 
+elogind = dependency('systemd', 'libelogind', required: get_option('elogind'))
+if elogind.found()
+  conf.set('HAVE_LOGIND' , '1')
+endif
+
 supported_build = get_option('supported_build').disable_auto_if(not tag).allowed()
 if supported_build
     conf.set('SUPPORTED_BUILD', '1')
@@ -702,6 +707,7 @@ summary({
   'dbus_socket_address': get_option('dbus_socket_address'),
   'vendor_ids_dir': vendor_ids_dir,
   'docs': build_docs,
+  'elogind': elogind,
   'gnutls': gnutls,
   'introspection': introspection.allowed(),
   'libblkid': libblkid,
diff --git a/meson_options.txt b/meson_options.txt
index 40103c0..bf8e964 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -39,6 +39,10 @@ option('efi_os_dir',
   type: 'string',
   description: 'the hardcoded name of OS directory in ESP, e.g. fedora',
 )
+option('elogind',
+  type: 'feature',
+  description: 'elogind support',
+)
 option('firmware-packager',
   type: 'boolean',
   value: true,
diff --git a/plugins/logind/meson.build b/plugins/logind/meson.build
index f07118c..257b8f4 100644
--- a/plugins/logind/meson.build
+++ b/plugins/logind/meson.build
@@ -1,4 +1,4 @@
-if libsystemd.found()
+if libsystemd.found() or elogind.found()
 cargs = ['-DG_LOG_DOMAIN="FuPluginLogind"']
 plugins += {meson.current_source_dir().split('/')[-1]: true}
