--- a/atom.sh
+++ b/atom.sh
@@ -123,31 +123,20 @@
   SCRIPT=$(readlink -f "$0")
   USR_DIRECTORY=$(readlink -f $(dirname $SCRIPT)/..)
 
-  case $CHANNEL in
-    beta)
-      ATOM_PATH="$USR_DIRECTORY/share/atom-beta/atom"
-      ;;
-    nightly)
-      ATOM_PATH="$USR_DIRECTORY/share/atom-nightly/atom"
-      ;;
-    dev)
-      ATOM_PATH="$USR_DIRECTORY/share/atom-dev/atom"
-      ;;
-    *)
-      ATOM_PATH="$USR_DIRECTORY/share/atom/atom"
-      ;;
-  esac
+  export ATOM_RESOURCE_PATH="{{ATOM_RESOURCE_PATH}}"
+  export LOCAL_GIT_DIRECTORY="{{ATOM_PREFIX}}/usr"
+  ATOM_PATH="{{ATOM_PATH}}"
 
   : ${TMPDIR:=/tmp}
 
   [ -x "$ATOM_PATH" ] || ATOM_PATH="$TMPDIR/atom-build/Atom/atom"
 
   if [ $EXPECT_OUTPUT ]; then
-    "$ATOM_PATH" --executed-from="$(pwd)" --pid=$$ "$@"
+    "$ATOM_PATH" --app="$ATOM_RESOURCE_PATH" --executed-from="$(pwd)" --pid=$$ "$@"
     exit $?
   else
     (
-    nohup "$ATOM_PATH" --executed-from="$(pwd)" --pid=$$ "$@" > "$ATOM_HOME/nohup.out" 2>&1
+    nohup "$ATOM_PATH" --app="$ATOM_RESOURCE_PATH" --executed-from="$(pwd)" --pid=$$ "$@" > "$ATOM_HOME/nohup.out" 2>&1
     if [ $? -ne 0 ]; then
       cat "$ATOM_HOME/nohup.out"
       exit $?
