--- a/plymouth.c
+++ b/plymouth.c
@@ -249,16 +249,6 @@ int rc_plugin_hook(RC_HOOK hook, const char *name)
         }
         break;
 
-    case RC_HOOK_SERVICE_STOP_NOW:
-        if(!ply_message("Stopping service", name))
-            rv = 1;
-        break;
-
-    case RC_HOOK_SERVICE_START_NOW:
-        if(!ply_message("Starting service", name))
-            rv = 1;
-        break;
-
     case RC_HOOK_SERVICE_START_OUT:
         /* Start Plymouth daemon if not yet started and tell we have rw access
          * to /var/... */